src/DDTrace/Integrations/Mysqli.php

```diff
@@ -0,0 +1,425 @@
+<?php
+
+namespace DDTrace\Integrations;
+
+use DDTrace\Span;
+use DDTrace\Tags;
+use DDTrace\Types;
+use DDTrace\Util\ObjectKVStore;
+use OpenTracing\GlobalTracer;
+
+class Mysqli
+{
+    public static function load()
+    {
+        if (!extension_loaded('ddtrace')) {
+            trigger_error('ddtrace extension required to load mysqli integrations.', E_USER_WARNING);
+            return;
+        }
+        if (!extension_loaded('mysqli')) {
+            trigger_error('mysqli is not loaded and cannot be instrumented', E_USER_WARNING);
+            return;
+        }
+
+        // mysqli mysqli_connect ([ string $host = ini_get("mysqli.default_host")
+        //      [, string $username = ini_get("mysqli.default_user")
+        //      [, string $passwd = ini_get("mysqli.default_pw")
+        //      [, string $dbname = ""
+        //      [, int $port = ini_get("mysqli.default_port")
+        //      [, string $socket = ini_get("mysqli.default_socket") ]]]]]] )
+        dd_trace('mysqli_connect', function () {
+            $args = func_get_args();
+            $scope = Mysqli::initScope('mysqli_connect', 'mysqli_connect');
```

Comment:
Unfortunately this is not possible because within the the `ddtrace` callback the scope (aka `self`) is not the `Mysqli` integration class. It is the `mysqli` object.

Reply:
You're right! In normal circumstances using `self` in a closure isn't a problem, but I can see how that won't work in this case for `dd_trace()`. :)